isofit/__main__.py

```diff
@@ -0,0 +1,332 @@
+"""ISOFIT command line interface."""
+
+
+import logging
+from types import SimpleNamespace
+
+import click
+
+import isofit
+
+################################################################################
+# CLI entrypoint
+
+
+@click.group(invoke_without_command=True)
+@click.pass_context
+@click.option("-v", "--version", help="Print the current version", is_flag=True)
+@click.option("-p", "--path", help="Print the installation path", is_flag=True)
+def cli(ctx, version, path):
+    """\
+    This houses the subcommands of ISOFIT
+    """
+    if ctx.invoked_subcommand is None:
+        if version:
+            click.echo(isofit.__version__)
+
+        if path:
+            click.echo(__path__[0])
+
+
+################################################################################
+# Subcommand: run
+
+
+@cli.command()
+@click.argument("config_file")
+@click.option(
+    "-l",
+    "--level",
+    help="Log level",
+    type=click.Choice(
+        ["DEBUG", "INFO", "WARNING", "ERROR", "EXCEPTION"], case_sensitive=True
+    ),
+    default="INFO",
+)
+def run(config_file, level):
+
+    """Execute ISOFIT core"""
+
+    # Delay import to keep CLI startup fast
+    from isofit.core.isofit import Isofit
+
+    click.echo(f"Running ISOFIT(config_file={config_file!r}, level={level})")
+
+    logging.basicConfig(format="%(message)s", level=level)
+    Isofit(config_file=config_file, level=level).run()
+
+    click.echo("Done")
+
+
+################################################################################
+# Subcommand: HRRR_to_modtran
+
+
+@cli.command()
+@click.argument("config_file")
+def HRRR_to_modtran(**kwargs):
+
+    """Add HRRR profiles to MODTRAN"""
+
+    # Delay import to keep CLI startup fast
+    from isofit.utils.add_HRRR_profiles_to_modtran_config import (
+        HRRR_to_MODTRAN_profiles,
+    )
+
+    click.echo("Running adding HRRR profiles to MODTRAN")
+
+    HRRR_to_MODTRAN_profiles(**kwargs)
+
+    click.echo("Done")
+
+
+################################################################################
+# Subcommand: analytical_line
+
+
+@cli.command()
+@click.argument("rdn_file")
+@click.argument("loc_file")
+@click.argument("obs_file")
+@click.argument("isofit_dir")
+@click.option("--isofit_config", type=str, default=None)
+@click.option("--segmentation_file", help="TODO", type=str, default=None)
+@click.option("--n_atm_neighbors", help="TODO", type=int, default=20)
+@click.option("--n_cores", help="TODO", type=int, default=-1)
```

Comment:
Good to know! I personally like the explicit declaration for each subcommand (especially if we ever break them into independent files), but I could be swayed 🙂

Reply:
Each declaration is an opportunity for slight differences to creep in. Reusing options provides a consistent set of flags across subcommands. However, reusing options does not *guarantee* consistence since ultimately it is up to the subcommand's code to actually *be* consistent.

Some problems in the current code:

3 different log file flags:

```
isofit/__main__.py:@click.option("--log_file")
isofit/__main__.py:@click.option("--logfile")
isofit/__main__.py:@click.option("--logfile", help="TODO", type=str, default=None)
```

3 different logging level flags:

```
isofit/__main__.py:@click.option("--logging_level", default="INFO")
isofit/__main__.py:@click.option("--loglevel", default="INFO")
isofit/__main__.py:@click.option("--loglevel", help="TODO", type=str, default="INFO"
```

3 different flags for specifying the number of cores to use:

```
isofit/__main__.py:@click.option("--n_cores", help="TODO", type=int, default=-1)
isofit/__main__.py:@click.option("--n_cores", type=int, default=-1)
isofit/__main__.py:@click.option("--n_cores", type=int, default=1)
```

Some of these differences don't matter, but some do.

Reply:
Opened #425 to address this.